py/desispec/test/test_photo.py

```diff
@@ -7,9 +7,9 @@
 
 import numpy as np
 from desispec.io.photo import gather_targetphot, gather_tractorphot, gather_targetdirs
+from desispec.io.meta import get_desi_root_readonly 
 
-if 'NERSC_HOST' in os.environ and \
-        os.getenv('DESI_SPECTRO_DATA') == '/global/cfs/cdirs/desi/spectro/data':
+if 'NERSC_HOST' in os.environ and os.getenv('DESI_SPECTRO_DATA') == os.path.join(desi_root, 'spectro', 'data'):
```

Comment:
This line fails at NERSC because you are using `desi_root` at the package level, before that variable is defined.  It isn't crashing on the github tests because `'NERSC_HOST' in os.environ` evaluates to False and it skips the rest of the checks on this line.

Reply:
Ugh. I had this correct at first but then broke it to fix a Github actions failure and didn't retest at NERSC. Should be fixed in https://github.com/desihub/desispec/pull/2154/commits/d4e3ef9d40c0bed27ce785470586da54888074b9.